star-dash/star-dash/GameEngine/Systems/Complex/Movement/FlyingModule.swift

```diff
@@ -112,6 +112,18 @@ class FlyingModule: MovementModule {
         spriteSystem.setSize(of: entityId, to: PhysicsConstants.Dimensions.plane)
     }
 
+    func cancelMovement(for entityId: EntityId) {
```

Comment:
i think its better to just call cancelFlying(for: entityId) to avoid avoid eduplication

Reply:
The problem with that is that it also calls will also call endAnimation which for some reason causes a delay before the dying animation starts. Same for movement.

To resolve it I added an endAnimation flag.